pykwasm/src/pykwasm/kdist/wasm-semantics/wasm.md

```diff
@@ -1776,24 +1719,19 @@ The value of a global gets copied when it is imported.
          <moduleInst>
            <modIdx> CUR </modIdx>
            <memIds> IDS => #saveId(IDS, OID, 0) </memIds>
-           <memAddrs> .Map => 0 |-> ADDR </memAddrs>
+           <memAddrs> .List => ListItem(ADDRS[#ContextLookup(IDS', TFIDX)]) </memAddrs>
            ...
          </moduleInst>
          <moduleRegistry> ... MOD |-> MODIDX ... </moduleRegistry>
          <moduleInst>
            <modIdx> MODIDX </modIdx>
            <memIds> IDS' </memIds>
-           <memAddrs> ... #ContextLookup(IDS' , TFIDX) |-> ADDR ... </memAddrs>
+           <memAddrs> ADDRS </memAddrs>
            <exports>  ... NAME |-> TFIDX                        ... </exports>
            ...
          </moduleInst>
-         <memInst>
-           <mAddr> ADDR </mAddr>
-           <mmax>  MAX  </mmax>
-           <msize> SIZE </msize>
-           ...
-         </memInst>
-       requires #limitsMatchImport(SIZE, MAX, LIM)
+         <mems> MEMS </mems>
+       requires #let memInst(MAX, SIZE, _) = MEMS[{ADDRS[#ContextLookup(IDS', TFIDX)]}:>Int] #in #limitsMatchImport(SIZE, MAX, LIM)
```

Comment:
Similar requirement here.

Also, in the same logic, we may need the additional requirement `#ContextLookup(IDS', TFIDX) <Int size(ADDRS)`.

Reply:
In all `requires` rules?